File: apollo-api/src/main/java/com/apollographql/apollo/api/internal/Mutator.java
package com.apollographql.apollo.api.internal;

/**
 * Represents an operation that accepts a single input argument, mutate it and returns no result
 *
 * @param <T> the type of the input to the operation
 */
public interface Mutator<T> {

  /**
   * Performs this operation on the given argument.
   *
   * @param t the input argument
   */
  void accept(T t);

}

File: apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/internal/Mutator.kt
package com.apollographql.apollo3.api.internal

/**
 * Represents an operation that accepts a single input argument, mutate it and returns no result
 *
 * @param <T> the type of the input to the operation
</T> */
interface Mutator<T> {
  /**
   * Performs this operation on the given argument.
   *
   * @param t the input argument
   */
  fun accept(t: T)
}
